dask/array/blockwise.py

```diff
@@ -197,7 +199,20 @@ def blockwise(func, out_ind, *args, **kwargs):
                         "adjust_chunks values must be callable, int, or tuple")
     chunks = tuple(chunks)
 
-    return Array(graph, out, chunks, dtype=dtype)
+    # Attempt to construct meta, fallback to Array's dtype constructor if it's
+    # a basic type
+    try:
+        arrays = args[::2]
+        arrays_empty = [arr._meta if hasattr(arr, '_meta') else np.empty((0,))
+                        for arr in arrays]
+        meta = func(*arrays_empty)
```

Comment:
This function probably needs all of the inputs, not just the arrays.  Maybe something like the following:

```python
args_meta = [arg._meta if hasattr(arg, '_meta') else arg for arg in args]
kwargs_meta = {k: v._meta if hasattr(v, '_meta') else v for k, v in kwargs.items()}
func = func(*args_meta, **kwargs_meta)
```

This happens because sometimes people put non-arrays in `map_blocks` or other `blockwise` operations like the following:

    x.map_blocks(func, y, 123, option='foo')

In these cases it's important to keep the extra args/keyword args

Reply:
I was aware my handling wasn't complete, but I wasn't sure what else was missing there, thanks for pointing that out. And to be honest, I wouldn't be surprised if your changes don't make it complete either, there are really many things that can happen in this `blockwise()` function.

Reply:
Yes, it's at the core of many calls and is the frequent target of many many feature requests.  Keeping it as simple as it is simple has been a challenge.

Reply:
(not that I think it's simple)

Regardless, thanks for being willing to dive into it.  I acknowledge that it's not trivial.

Reply:
I did the changes you suggested, it mostly works but breaks `test_argwhere_str` and `test_count_nonzero_str`. The issue is with passing `**kwargs`, after a lot of unsuccessful digging, I think what happens is the 'concatenate' argument is passed to `numpy.vectorize()` that doesn't expect it. I will come back to check this later.

Reply:
I'll try to take a look at this later today as well

Reply:
It's still difficult to handle all cases when we pass `kwargs`, I'm still working on fixing this.